daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -460,37 +411,30 @@ func (l *logStream) processEvent(events []wrappedEvent, unprocessedLine []byte,
 		}
 		line := unprocessedLine[:lineBytes]
 		unprocessedLine = unprocessedLine[lineBytes:]
-		if (len(events) >= maximumLogEventsPerPut) || (bytes+lineBytes+perEventBytes > maximumBytesPerPut) {
+		if (batch.Count() >= maximumLogEventsPerPut) || (batch.Size()+lineBytes+perEventBytes > maximumBytesPerPut) {
```

Comment:
Just thinking out loud here; wondering if we could make `batch.Add()` reject the event if the Size or Count limit was reached, something like (dummy code);

```go
err := batch.Add(event)

switch err := err.(type) {
case <maximum size reached>, <maximum logs reached>:
   l.publishBatch(batch)
case nil:
   unprocessedLine = unprocessedLine[lineBytes:]
default:
  // something bad happened :D
}
```
     



}

Reply:
That would certainly clean up some of these checked `Add()`s! I like the indication of the error and codification of the limit check in the batch itself, let me whip something up.